onboard/nav/avoidances.cpp

```diff
@@ -0,0 +1,164 @@
+/* Searches.cpp */
+
+#include "avoidances.hpp"
+#include "utilities.hpp"
+#include "stateMachine.hpp"
+
+#include <iostream>
+#include <cmath>
+#include <stack>
+
+/*************************************************************************/
+/* Avoidance Factory */
+/*************************************************************************/
+// Avoidance Factory to allow for creation of obstacle avoidance objects.
+// Allows for easy change from obstacle avoidance algorithm to another
+Avoidance* AvoidFactory( StateMachine* stateMachine, AvoidanceType type )
+{
+    Avoidance* avoid = nullptr;
+    switch (type)
+    {
+        case AvoidanceType::ORIGINAL:
+            avoid = new Original( stateMachine );
+            break;
+
+        case AvoidanceType::UNKNOWN:
+            std::cerr << "Unkown Search Type. Defaulting to original\n";
+            avoid = new Original( stateMachine );
+            break;
+    }
+    return avoid;
+}
+
+/*************************************************************************/
+/* Original Obstacle Avoidance Algorithm
+   When an obstacle is detected, the rover will drive around whichever way
+   CV decides is the fastest. If the tennisball is detected, the heirarchy has
+   two cases in which the rover will begin to turn to tennis ball:
+        1. the difference in distance (tennis ball is closer than obs)
+        2. Turning around obs is same direction to turn to tennis ball */
+/*************************************************************************/
+Original::~Original() {}
+
+/*****************************************************/
+/* Helpers */
+/*****************************************************/
+
+// Executes the logic for turning around an obstacle.
+// If the rover is turned off, it proceeds to Off.
+// If the tennis ball is detected, the rover proceeds to it.
+// If the obstacle is no longer detcted, proceed to drive around the obstacle.
+// Else continue turning around the obstacle.
+// ASSUMPTION: We assume that the obstacle is straight ahead of us,
+//             therefore we produce an underestimate for how close the waypoint is to the
+//             obstacle. This relies on using a path width no larger than what we can
+//             confidentally see to the side.
+//
+//             There is no rock that is more than 8 meters in diameter
+NavState Original::executeTurnAroundObs( Rover * mPhoebe, const rapidjson::Document& roverConfig )
```

Comment:
Make `mPhoebe` into `phoebe` because it is not a member. Please do this throughout the files and be consistent with the placement of `*` in `Rover* pheobe`. While I'd prefer `Rover* pheobe` to be consistent with the rest of our code, whichever you choose doesn't matter but it should be consistent inside the file at least